specs/simple-serialize.md

```diff
@@ -1,424 +1,108 @@
-      - [Container](#container-2)
-   + [Signed Roots](#signed-roots)
-* [Implementations](#implementations)
+## Typing
 
-## About
+### Basic types
 
-`SimpleSerialize` was first proposed by Vitalik Buterin as the serialization
-protocol for use in the Ethereum 2.0 Beacon Chain.
+* `uintN`: `N`-bit unsigned integer (where `N in [8, 16, 32, 64, 128, 256]`)
+* `bool`: 1-bit unsigned integer
 
-The core feature of `ssz` is the simplicity of the serialization with low
-overhead.
+### Composite types
 
-## Variables and Functions
+* **container**: ordered heterogenous collection of values
+    * key-pair curly braket notation `{}`, e.g. `{'foo': uint64, 'bar': bool}`
+* **tuple**: ordered fixed-length homogeneous collection of values
+    * angle braket notation `[N]`, e.g. `uint64[N]`
+* **list**: ordered variable-length homogenous collection of values
+    * angle braket notation `[]`, e.g. `uint64[]`
 
-| Term         | Definition                                                                                     |
-|:-------------|:-----------------------------------------------------------------------------------------------|
-| `little`     | Little endian.                                                                                 |
-| `byteorder`  | Specifies [endianness](https://en.wikipedia.org/wiki/Endianness): big endian or little endian. |
-| `len`        | Length/number of bytes.                                                                        |
-| `to_bytes`   | Convert to bytes. Should take parameters ``size`` and ``byteorder``.                           |
-| `from_bytes` | Convert from bytes to object. Should take ``bytes`` and ``byteorder``.                         |
-| `value`      | The value to serialize.                                                                        |
-| `rawbytes`   | Raw serialized bytes.                                                                          |
-| `deserialized_object` | The deserialized data in the data structure of your programming language.             |
-| `new_index`  | An index to keep track the latest position where the `rawbytes` have been deserialized.        |
+### Aliases
 
-## Constants
+For convenience we alias:
 
-| Constant          | Value | Definition                                                                            |
-|:------------------|:-----:|:--------------------------------------------------------------------------------------|
-| `LENGTH_BYTES`    |   4   | Number of bytes used for the length added before a variable-length serialized object. |
-| `SSZ_CHUNK_SIZE`  |  128  | Number of bytes for the chunk size of the Merkle tree leaf.                           |
+* `byte` to `uint8`
+* `bytes` to `byte[]`
+* `bytesN` to `byte[N]`
 
-## Overview
+## Serialization
 
-### Serialize/Encode
+We reccursively define the `serialize` function which consumes an object `value` (of the type specified) and returns a byte string of type `bytes`.
```

Comment:
> the language implementation might interpret it rather than defining the output as an SSZ type itself

That doesn't work great because we define `serialize` recursively, so the output of `serialize` itself needs to be an object of a well-defined type.

Reply:
I see. sounds good